api/app/hfi/pdf_data_formatter.py

```diff
@@ -0,0 +1,50 @@
+""" Marshals HFI result into structure that jinja can easily
+        iterate over for generating the daily PDF sheets
+"""
+from itertools import groupby
+import operator
+from typing import List, Mapping
+from app.schemas.hfi_calc import (DailyPDFData,
+                                  HFIResultResponse,
+                                  PlanningArea,
+                                  StationDaily,
+                                  StationPDFData,
+                                  WeatherStation)
+
+
+def response_2_daily_jinja_format(result: HFIResultResponse,
+                                  planning_area_dict: Mapping[int, PlanningArea],
+                                  station_dict: Mapping[int, WeatherStation]):  # pylint: disable=line-too-long
+    """ Marshals HFI result into structure that jinja can easily
+        iterate over for generating the daily PDF sheets
+     """
+
+    daily_pdf_data: List[DailyPDFData] = []
+    for area_result in result.planning_area_hfi_results:
+        fire_starts_range = result.planning_area_fire_starts[area_result.planning_area_id]
+        days_total = len(area_result.daily_results)
+        for j, daily_result in enumerate(area_result.daily_results):
+            dailies: List[StationDaily] = list(map(lambda x: x.daily, daily_result.dailies))
+            full_dailies: List[StationPDFData] = []
+            for daily in dailies:
+                station_data = station_dict[daily.code]
+                merged = daily.dict()
+                merged.update(station_data)
+                full_daily = StationPDFData(**merged)
+                full_dailies.append(full_daily)
+            fire_starts = fire_starts_range[j]
+            planning_area_name = planning_area_dict[area_result.planning_area_id].name
+            # TODO: Get planning area name, not just id
```

Comment:
I don't think this TODO is needed anymore, is it (I haven't tried generating a PDF myself but I thought you have PA names in there now)?

Reply:
Thanks, removed